src/main/java/org/opensearch/neuralsearch/processor/ExplanationResponseProcessor.java

```diff
@@ -99,16 +100,26 @@ public SearchResponse processResponse(
                     ExplanationDetails normalizationExplanation = combinedExplainDetail.getNormalizationExplanations();
                     ExplanationDetails combinationExplanation = combinedExplainDetail.getCombinationExplanations();
                     // Create normalized explanations for each detail
-                    Explanation[] normalizedExplanation = new Explanation[queryLevelExplanation.getDetails().length];
-                    for (int i = 0; i < queryLevelExplanation.getDetails().length; i++) {
-                        normalizedExplanation[i] = Explanation.match(
-                            // normalized score
-                            normalizationExplanation.getScoreDetails().get(i).getKey(),
-                            // description of normalized score
-                            normalizationExplanation.getScoreDetails().get(i).getValue(),
-                            // shard level details
-                            queryLevelExplanation.getDetails()[i]
-                        );
+                    List<Explanation> normalizedExplanation = new ArrayList<>(queryLevelExplanation.getDetails().length);
+                    int normalizationExplanationIndex = 0;
+                    for (Explanation queryExplanation : queryLevelExplanation.getDetails()) {
+                        // adding only explanations where this hit has matched
+                        if (Float.compare(queryExplanation.getValue().floatValue(), 0.0f) > 0) {
+                            Pair<Float, String> normalizedScoreDetails = normalizationExplanation.getScoreDetails()
+                                .get(normalizationExplanationIndex);
```

Comment:
There is no such case where this `get` will return null?
```
assert normalizedScoreDetails != null
```

Reply:
ack